rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -3141,6 +3158,46 @@ public void addNodeToken(String nodeUrl, String token) throws RMException {
         }
     }
 
+    @Override
+    public Set<String> getNodeTags(String nodeUrl) throws RMException {
+        if (allNodes.containsKey(nodeUrl)) {
+            RMNode rmNode = allNodes.get(nodeUrl);
+            return rmNode.getNodeTags();
+        } else {
+            throw new RMException("Unknown node: " + nodeUrl);
+        }
+    }
+
+    @Override
+    public Set<String> getNodesByTags(String tag) {
+        Set<String> result = new HashSet<>();
+        for (RMNode rmNode : this.allNodes.values()) {
+            if (rmNode.getNodeTags().contains(tag)) {
+                result.add(rmNode.getNodeURL());
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public Set<String> getNodesByTags(List<String> tags, boolean all) {
+        Set<String> result = new HashSet<>();
+        for (RMNode rmNode : this.allNodes.values()) {
+            if (all) {
+                if (rmNode.getNodeTags().containsAll(tags)) {
```

Comment:
Similarly, shouldn't there any exception thrown when no nodes could be selected ?

Reply:
Similarly, an empty list is returned.